pinot-tools/src/main/java/org/apache/pinot/tools/PinotTableRebalancer.java

```diff
@@ -0,0 +1,123 @@
+  private final boolean _dryRun;
+  private final int _minReplicasToKeepUpForNoDowntime;
+  private TableRebalancer.RebalancerStats _rebalancerStats;
+
+  /**
+   * Creates an instance of PinotTableRebalancer
+   * @param zkAddress zookeeper address of the cluster
+   * @param pinotClusterName pinot cluster name
+   * @param dryRun true if rebalancer should run in dry run mode (just return final ideal state), false otherwise
+   * @param noDowntime true if rebalancer should run without downtime, false otherwise
+   * @param includeConsuming true if rebalancer should include consuming segments
+   *                         while rebalancing, false otherwise
+   * @param minReplicasToKeepUpForNoDowntime applicable for noDowntime approach
+   */
+  public PinotTableRebalancer(final String zkAddress, final String pinotClusterName, final boolean dryRun,
+      final boolean noDowntime, final boolean includeConsuming, final int minReplicasToKeepUpForNoDowntime) {
+    super(zkAddress, pinotClusterName);
+    _noDowntime = noDowntime;
+    _includeConsuming = includeConsuming;
+    _dryRun = dryRun;
+    _minReplicasToKeepUpForNoDowntime = minReplicasToKeepUpForNoDowntime;
+  }
+
+  public boolean rebalance(final String tableName, final String tableType)
+      throws Exception {
+    Preconditions.checkArgument(tableName != null && tableName.length() > 0, "Expecting a valid table name");
+    Preconditions.checkArgument(
+        tableType != null && (tableType.equalsIgnoreCase("OFFLINE") || tableType.equalsIgnoreCase("REALTIME")),
+        "Expecting a valid table type (OFFLINE or REALTIME");
+
+    final Configuration rebalanceUserConfig = new PropertiesConfiguration();
+    rebalanceUserConfig.addProperty(RebalanceUserConfigConstants.DRYRUN, _dryRun);
+    rebalanceUserConfig.addProperty(RebalanceUserConfigConstants.INCLUDE_CONSUMING, _includeConsuming);
+    rebalanceUserConfig.addProperty(RebalanceUserConfigConstants.DOWNTIME, !_noDowntime);
+
+    if (_noDowntime) {
+      rebalanceUserConfig.addProperty(RebalanceUserConfigConstants.MIN_REPLICAS_TO_KEEPUP_FOR_NODOWNTIME,
+          _minReplicasToKeepUpForNoDowntime);
+    }
+
+    final TableType tabType = TableType.valueOf(tableType);
+    final TableConfig tableConfig = getTableConfig(tableName, tabType);
+    final TableRebalancer tableRebalancer = new TableRebalancer(helixManager, helixAdmin, clusterName);
+    final RebalanceSegmentStrategyFactory rebalanceSegmentStrategyFactory =
+        new RebalanceSegmentStrategyFactory(helixManager);
+    final RebalanceSegmentStrategy rebalanceSegmentsStrategy =
+        rebalanceSegmentStrategyFactory.getRebalanceSegmentsStrategy(tableConfig);
+    final RebalanceResult rebalanceResult =
+        tableRebalancer.rebalance(tableConfig, rebalanceSegmentsStrategy, rebalanceUserConfig);
+
+    if (rebalanceResult.getStatus() == RebalanceResult.RebalanceStatus.FAILED) {
+      System.out.println("Failed to rebalance table: " + tableName);
```

Comment:
Are we not using `LOGGER` in tools package?

Reply:
Initially didn't use it for manual testing. Switched to LOGGER now. It is being used in some of the other tools code as well.